Samples/PublicSamples/EchoBot/ADF/bicep/OMS.bicep

```diff
@@ -827,148 +827,144 @@ resource updateConfigWindows 'Microsoft.Automation/automationAccounts/softwareUp
-                    scheduledTransferPeriod: 'PT1M'
-                    samplingFrequencyInSeconds: 30
-                    counterSpecifiers: [
-                        '\\Memory\\Available Bytes'
-                        '\\Memory\\Committed Bytes'
-                        '\\Processor(_Total)\\% Processor Time'
-                        '\\LogicalDisk(*)\\% Free Space'
-                        '\\LogicalDisk(_Total)\\Free Megabytes'
-                        '\\PhysicalDisk(_Total)\\Avg. Disk Queue Length'
-                    ]
-                    streams: [
-                        'Microsoft-Perf'
-                    ]
-                }
-                {
-                    name: 'appTeamExtraCounters'
-                    streams: [
-                        'Microsoft-Perf'
-                    ]
-                    scheduledTransferPeriod: 'PT5M'
-                    samplingFrequencyInSeconds: 30
-                    counterSpecifiers: [
-                        '\\Process(_Total)\\Thread Count'
-                    ]
-                }
-            ]
-            extensions: [
-                {
-                    name: 'Microsoft-VMInsights-Health'
-                    streams: [
-                        'Microsoft-HealthStateChange'
-                    ]
-                    extensionName: 'HealthExtension'
-                    extensionSettings: {
-                        schemaVersion: '1.0'
-                        contentVersion: ''
-                        healthRuleOverrides: [
-                            {
-                                scopes: [
-                                    '*'
-                                ]
-                                monitors: [
-                                    'root'
-                                ]
-                                monitorConfiguration: {}
-                                alertConfiguration: {
-                                    isEnabled: true
-                                }
-                            }
-                        ]
-                    }
-                    inputDataSources: [
-                        'VMHealthPerfCounters'
-                    ]
-                }
-            ]
-        }
-        destinations: {
-            logAnalytics: [
-                {
-                    workspaceResourceId: OMS.id
-                    name: 'LogAnalyticsWorkspace'
-                }
-            ]
-        }
-        dataFlows: [
-            {
-                streams: [
-                    'Microsoft-HealthStateChange'
-                    'Microsoft-Perf'
-                    'Microsoft-Syslog'
-                    'Microsoft-WindowsEvent'
-                ]
-                destinations: [
-                    'LogAnalyticsWorkspace'
-                ]
-            }
-        ]
-    }
-}
+// resource VMInsights 'Microsoft.Insights/dataCollectionRules@2022-06-01' = if (bool(Extensions.VMInsights)) {
```

Comment:
So I guess this could go entirely?

Reply:
There are a few places that have commented out code for a few different reasons. The main reason you see commented out code in the bicep templates is because we are not currently using that additional feature or configuration. But it can be useful for someone else if they want to implement it. As an example, someone might want to implement the Bastion service for managing how they access the VMs. That code is there but commented out. This specific example I think a breaking change was introduced and the queries need to be updated. It is something that should be updated in a future PR, but not necessary for V2 IMO. I left it in there in case someone from the community needed that feature and they would be able to uncomment, test and modify.